glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -1211,10 +1289,17 @@ where
     // Being used when all cluster connections are unavailable.
     fn reconnect_to_initial_nodes(inner: Arc<InnerCore<C>>) -> impl Future<Output = ()> {
         let inner = inner.clone();
-        async move {
+        let cluster_params = match inner.get_cluster_param(|params| params.clone()) {
+            Ok(params) => params,
+            Err(err) => {
+                warn!("Failed to get cluster params: {}", err);
+                return async {}.boxed();
```

Comment:
Change the function to return `impl Future<Output = RedisResult<()>`, and instead of logging the error return it

Reply:
Major changes in the state machine, separate PR.